torch/csrc/jit/interpreter.cpp

```diff
@@ -934,7 +951,8 @@ struct InterpreterStateImpl : c10::intrusive_ptr_target {
   }
 
   void enterFrame(const Code& code, size_t base_pointer) {
-    frames.emplace_back(Frame{code.pImpl, 0, base_pointer});
+    frames.emplace_back(
+        Frame{code.pImpl, 0, base_pointer, Frame::num_frames++});
```

Comment:
I am not a huge fan of introducing atomics along all call boundaries just to do profiling.  It puts logic related to something that happens rarely in the fast path.

Reply:
Can the frame_id be assigned the first time a profile node is run in the frame?